pkg/controller/bundlec/bundle_sync_task.go

```diff
@@ -329,50 +329,13 @@ func (st *bundleSyncTask) handleProcessResult(retriable bool, processErr error)
 		var failedResources []smith_v1.ResourceName
 		retriableResourceErr := true
 		for _, res := range st.bundle.Spec.Resources { // Deterministic iteration order
-			blockedCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceBlocked, Status: smith_v1.ConditionFalse}
-			inProgressCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceInProgress, Status: smith_v1.ConditionFalse}
-			readyCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceReady, Status: smith_v1.ConditionFalse}
-			errorCond := smith_v1.ResourceCondition{Type: smith_v1.ResourceError, Status: smith_v1.ConditionFalse}
-
-			if resInfo, ok := st.processedResources[res.Name]; ok {
-				// Resource was processed
-				switch resStatus := resInfo.status.(type) {
-				case resourceStatusDependenciesNotReady:
-					blockedCond.Status = smith_v1.ConditionTrue
-					blockedCond.Reason = smith_v1.ResourceReasonDependenciesNotReady
-					blockedCond.Message = fmt.Sprintf("Not ready: %q", resStatus.dependencies)
-				case resourceStatusInProgress:
-					inProgressCond.Status = smith_v1.ConditionTrue
-				case resourceStatusReady:
-					readyCond.Status = smith_v1.ConditionTrue
-				case resourceStatusError:
-					errorCond.Status = smith_v1.ConditionTrue
-					errorCond.Message = resStatus.err.Error()
-					if resStatus.isRetriableError {
-						errorCond.Reason = smith_v1.ResourceReasonRetriableError
-						inProgressCond.Status = smith_v1.ConditionTrue
-					} else {
-						errorCond.Reason = smith_v1.ResourceReasonTerminalError
-					}
-					failedResources = append(failedResources, res.Name)
-					retriableResourceErr = retriableResourceErr && resStatus.isRetriableError // Must not continue if at least one error is not retriable
-				default:
-					blockedCond.Status = smith_v1.ConditionUnknown
-					inProgressCond.Status = smith_v1.ConditionUnknown
-					readyCond.Status = smith_v1.ConditionUnknown
-					errorCond.Status = smith_v1.ConditionTrue
-					errorCond.Reason = smith_v1.ResourceReasonTerminalError
-					errorCond.Message = fmt.Sprintf("internal error - unknown resource status type %T", resInfo.status)
-					failedResources = append(failedResources, res.Name)
-					retriableResourceErr = false
-				}
-			} else {
-				// Resource was not processed
-				blockedCond.Status = smith_v1.ConditionUnknown
-				inProgressCond.Status = smith_v1.ConditionUnknown
-				readyCond.Status = smith_v1.ConditionUnknown
-				errorCond.Status = smith_v1.ConditionUnknown
+			blockedCond, inProgressCond, readyCond, errorCond := st.resourceConditions(res)
```

Comment:
This change helped reduce the cyclomatic complexity for `handleProcessResult()`

Reply:
Nice one! Not because the linter is happy now, but because it reads much better.